src/main/java/com/dungeoncrawler/Javiarenka/character/Monster.java

```diff
@@ -23,6 +25,21 @@ public void setDamageStrength(int damageStrength) {
 
     @Override
     void attack(Character hero) {
-        hero.setHp(hero.getHp() - damageStrength);
+        Hero attackedHero = (Hero)hero;
+
+        if (attackedHero.getEquippedArmor().getDamageReduction() <= damageStrength) {
+            attackedHero.setHp(attackedHero.getHp() + attackedHero.getEquippedArmor().getDamageReduction() - damageStrength);
+        } else {
+            attackedHero.setHp(attackedHero.getHp());
+        }
+
+        Random random = new Random();
+        int hitPossibility = random.nextInt(100);
+
+        if (hitPossibility >= attackedHero.getEquippedArmor().getChanceToHitReduction()) {
+            attack(attackedHero);
```

Comment:
Teraz mamy sytuację, w której metoda attack jest wywoływana rekurencyjnie, także możliwe,, że takich ataków odpali się kilkadziesiąt. Ten warunek przeniósłbym na początek metody i zmienił go na coś w stylu:
`if (hitPossibility <= attackedHero.getEquippedArmor().getChanceToHitReduction()) {
return }`

Reply:
Poprawiłam. Chociaż teraz tak sobie myślę, że trzeba by zmienić nazwę zmiennej, bo ona sugeruje, że to redukcja szansy na trafienie, a metoda jest napisana jakby to była szansa na trafienie. Mam rację, czy mi się już pogmatwało?

Reply:
Wydaje mi się, że 
'''if (hitPossibility < attackedHero.getEquippedArmor().getChanceToHitReduction()) { return }'''
jest ok, bo jeżeli wyrzucona liczba jest mniejsza od współczynnika zbroi, to nie trafiasz i wychodzisz z metody ataku.